### ISSUE TITLE
update is "rejected by policy" for new branch policies

### ISSUE THREAD

**Issue by ceciliasharp** [microsoft/terraform-provider-azuredevops#569]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

azuredevops_branch_policy_min_reviewers
azuredevops_branch_policy_status_check
azuredevops_branch_policy_work_item_linking

### Terraform Configuration Files

```hcl

resource "azuredevops_branch_policy_min_reviewers" "p" {
  project_id = data.azuredevops_project.p.id

  enabled  = true
  blocking = true

  settings {
    reviewer_count     = 1
    submitter_can_vote = false
    last_pusher_cannot_approve = false
    allow_completion_with_rejects_or_waits = false
    on_push_reset_all_votes = true

    scope {
      repository_id  = data.azuredevops_git_repository.r.id
      repository_ref = data.azuredevops_git_repository.r.default_branch
      match_type     = "Exact"
    }

  }
}

resource "azuredevops_branch_policy_work_item_linking" "p" {
  project_id = data.azuredevops_project.p.id

  enabled  = true
  blocking = true

  settings {

    scope {
      repository_id  = data.azuredevops_git_repository.r.id
      repository_ref = data.azuredevops_git_repository.r.default_branch
      match_type     = "Exact"
    }

  }
} 

resource "azuredevops_branch_policy_comment_resolution" "p" {
  project_id = data.azuredevops_project.p.id

  enabled  = true
  blocking = true

  settings {

    scope {
      repository_id  = data.azuredevops_git_repository.r.id
      repository_ref = data.azuredevops_git_repository.r.default_branch
      match_type     = "Exact"
    }

  }
}

```

### Debug Output

│ Error: Error creating policy in Azure DevOps: The update is rejected by policy.
│
│   with azuredevops_branch_policy_comment_resolution....

### Behavior

If an repo in already have manually added branch policies (via the portal), this errors will occur. All three terraform resources will get the same error. If going into the portal and turn the polices off before applying the terraform it works and also if changing in the portal after terraform apply, and then apply after the change also works. Only when there is a pre-exisiting branch policy the apply will fail with an error about rejected by policy. I expect that it should be possible to apply terraform handling of the branches policies even if anyone have manually editing it from the portal before applying.

**Comment by xuzhang3** [comment]
@ceciliasharp  It's not recommend  to mange the resource by portal and Terraform at the same time.  You can import the existed policies by `terraform import`
Ref: https://registry.terraform.io/providers/microsoft/azuredevops/latest/docs/resources/branch_policy_work_item_linking#import

**Comment by ceciliasharp** [comment]
the problem is that I have 100+ repos that we are moving over to handle this with terraform so it's not a "both situation" it's "from portal to terraform"-handled. 60% of these repos have allready configurations today. I did solve it by manually remove the policy. I can't see that importing them oneby one all would have taken med shorter time than that (and the policy resource is declared as a foreach loop is that even possible to import?). 
I did only wish that if there was a policy present the terraform could override it but knowing how terraform works, that a policy is its own identity and that it can't be more than one policy; I understand if this can't be fixed.
